### PR TITLE
style: Enforce code style across all supported languages using Prettier

### PR DESCRIPTION
**Pull request by jamilbk**
This ensure that we run prettier across all supported filetypes to check for any formatting / style inconsistencies. Previously, it was only run for files in the website/ directory using a deprecated pre-commit plugin.

The benefit to keeping this in our pre-commit config is that devs can optionally run these checks locally with `pre-commit run --config .github/pre-commit-config.yaml`.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -186,14 +185,14 @@ jobs:
           docker compose up -d relay-1 relay-2 --no-build
           docker compose up -d gateway --no-build
           docker compose up -d client --no-build
-      - name: 'Performance test: ${{ matrix.test_name }}'
+      - name: "Performance test: ${{ matrix.test_name }}"
```
Comment (bmanifold) [inline .github/workflows/ci.yml:188]: Just wanted to confirm that changing this from `'` to `"` isn't going to change any of the expansion/interpolation for the `${{ }}` in the string?

**Review by bmanifold — APPROVED** [review]
LGTM.  One small question

File: rust/gui-client/pnpm-lock.yaml
Comment (thomaseizinger) [inline rust/gui-client/pnpm-lock.yaml:1]: I think we may want to exclude this one because it is generated?

File: .prettierignore [hunk 2]
```diff
@@ -0,0 +1,12 @@
+elixir/_build
+elixir/deps
+elixir/apps/web/assets
+rust/target
```
Comment (thomaseizinger) [inline .prettierignore:4]:
```suggestion
rust
```

We have our own formatter :)

File: .github/workflows/ci.yml (hunk 1 above)
Comment (thomaseizinger) [reply .github/workflows/ci.yml:188]: Seems to be working fine: https://github.com/firezone/firezone/actions/runs/11804416568/job/32884861190?pr=7322

**Review by thomaseizinger — APPROVED** [review]
LGTM just one concern

File: .github/workflows/ci.yml (hunk 1 above)
Comment (jamilbk) [reply .github/workflows/ci.yml:188]: In GitHub expressions there is a difference between `'` and `"`, but in YAML there isn't (AFAIK).

File: .prettierignore (hunk 2 above)
Comment (jamilbk) [reply .prettierignore:4]: Hmmm but there are typescript, markdown and other files in here. Maybe I should just exclude `*.rs` and `*.toml`?

Comment (jamilbk) [reply .prettierignore:4]: prettier ignores rust code anyway - it's not supported natively.

File: rust/gui-client/pnpm-lock.yaml
Comment (jamilbk) [reply rust/gui-client/pnpm-lock.yaml:1]: 👍

File: .github/workflows/ci.yml (hunk 1 above)
Comment (bmanifold) [reply .github/workflows/ci.yml:188]: According to the spec there is a difference between the two:
https://yaml.org/spec/1.2.2/#single-quoted-style
https://yaml.org/spec/1.2.2/#731-double-quoted-style

(Another ref about yaml quotes: https://www.yaml.info/learn/quote.html)

I'm guessing it's probably not going to make a difference here, but just wanted to make sure.

**Pull request merged**